### PR TITLE
Fix above_stake_threshold and add genesis leadership selection test

### PR DESCRIPTION
**Pull request by edolstra**
This code is primarily to experiment with the phi function, so a test is not really an ideal place for it. Might be better as an example?

### CODE REVIEW

**Comment by vincenthz** [comment]
Maybe best to see this as an executable in a separate crate ?
not sure what's the best way to access some of the function like the `create_fake_ledger`

**Comment by edolstra** [comment]
Apparently it's possible to disable a test by default by marking it with `#[ignore]`. That sounds like a good option here.

File: chain-time/src/timeframe.rs [hunk 1]
```diff
@@ -6,7 +6,7 @@ use std::time::{Duration, SystemTime};
 /// The slots are not comparable to others slots made on a
 /// different time frame
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Slot(pub(crate) u64);
+pub struct Slot(pub u64);
```
Comment (NicolasDP) [inline chain-time/src/timeframe.rs]: does it need to be public? I think this value, the only way we should get it, should be through the `TimeFrame` object. It's a bit annoying but this is the only way to guarantee this object is not constructed with the wrong value, or with value that does not make sense in a given timeframe.

Comment (edolstra) [reply chain-time/src/timeframe.rs]: I've fixed the test to use `ledger.settings.era` so it doesn't need to construct `Slot` anymore.

**Pull request merged**